File: AdHoc/UniquePaths.py
class Solution:
    def uniquePaths(self, A: int, B: int) -> int:
        if A==1 or B==1:
            return 1
        if A==2 or B==2:
            return max(A,B)
        ans = 1
        for i in range(max(A,B), A+B-1):
            ans *= i
            ans //= (i-max(A,B)+1)
        return ans

# LC 62 Medium

File: AdHoc/UniquePaths.cpp
// LC 62
long long int comb(int m,int n)
{
     long long int res = 1, i;
     int a=(m>n)? m: n;
     for (i=m+n-2;i>=a; i--) 
         res *= i; 
    return res; 
}

long long int fact(int n)
{
    long long int res=1,i;
    for(i=2;i<=n;i++)
        res*=i;
    return res;
}

int Solution::uniquePaths(int m, int n) {
    long long int a,k=comb(m,n);
    if(m<n)
        a=fact(m-1); // since m is smaller
    else
        a=fact(n-1); // else as n is smaller
    return k/a;
}

